fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/RankedSetIndexHelper.java

```diff
@@ -50,6 +54,54 @@
 public class RankedSetIndexHelper {
     public static final Tuple COMPARISON_SKIPPED_SCORE = Tuple.from(Comparisons.COMPARISON_SKIPPED_BINDING);
 
+    public static final RankedSet.HashFunction MURMUR3_HASH_FUNCTION = new GuavaHashFunction(Hashing.murmur3_32());
+
+    /**
+     * Use {@code com.google.common.hash.HashFunction}s as {@code RankedSet.HashFunction}s.
+     */
+    public static class GuavaHashFunction implements RankedSet.HashFunction {
+        @Nonnull
+        private final HashFunction guava;
+
+        public GuavaHashFunction(HashFunction guava) {
+            this.guava = guava;
+        }
+
+        @Override
+        public int hash(byte[] key) {
+            return guava.hashBytes(key).asInt();
+        }
+    }
+
+    /**
+     * Known hash functions available as index options.
+     */
+    public enum HashFunctionNames {
+        JDK(RankedSet.JDK_ARRAY_HASH),
+        CRC(RankedSet.CRC_HASH),
+        MURMUR3(MURMUR3_HASH_FUNCTION);
+
+        private final RankedSet.HashFunction hashFunction;
+
+        HashFunctionNames(RankedSet.HashFunction hashFunction) {
+            this.hashFunction = hashFunction;
+        }
+
+        public RankedSet.HashFunction getHashFunction() {
+            return hashFunction;
+        }
+    }
+
+    public static int getNLevels(@Nonnull Index index) {
+        String nlevelsOption = index.getOption(IndexOptions.RANK_NLEVELS);
+        return nlevelsOption == null ? RankedSet.DEFAULT_LEVELS : Integer.parseInt(nlevelsOption);
+    }
+
+    public static RankedSet.HashFunction getHashFunction(@Nonnull Index index) {
+        String hashFunctionOption = index.getOption(IndexOptions.RANK_HASH_FUNCTION);
+        return hashFunctionOption == null ? RankedSet.DEFAULT_HASH_FUNCTION : HashFunctionNames.valueOf(hashFunctionOption).getHashFunction();
```

Comment:
I suppose using a registry rather than an enum is more idiomatic (e.g., `FunctionKeyExpression`s) and allows a user to provide their own implementation. I think that switching to a registry in the future is compatible with this current approach, though, so maybe it's fine as is for now. (Perhaps there should be an Issue about it.)

Reply:
Yes. It's also possible that some additional options will require more state.

For example, if we wanted one that seeded Murmur3 from, say, the leaderboard time window id instead of always using 0.

Reply:
I didn’t make it extensible, but as a top level regular class it is now maybe a bit more registry like

Reply:
Yeah, and even though it's not extensible now, it looks like the kind of thing that could be made a registry if/when needed.